### PR TITLE
Add drag and drop for disks

### PR DESCRIPTION
**Pull request by whscullin**

### CODE REVIEW

File: js/components/util/files.ts [hunk 1]
```diff
@@ -218,27 +215,35 @@ export const loadHttpNibbleFile = async (
...
     const data = await loadHttpFile(url, onProgress);
     const { name, ext } = getNameAndExtension(url);
-    if (includes(DISK_FORMATS, ext)) {
-        if (data.byteLength >= 800 * 1024) {
-            if (includes(BLOCK_FORMATS, ext)) {
-                smartPort.setBinary(number, name, ext, data);
+    unknownStorage.setBinary(number, name, ext, data);
+};
+
+export class UnknownStorage implements MassStorage<unknown> {
```
Comment (whscullin) [inline js/components/util/files.ts:228]: Naming is hard.

File: js/components/Drives.tsx [hunk 2]
```diff
@@ -71,27 +73,31 @@ export const Drives = ({ cpu, io, sectors, enhanced, ready }: DrivesProps) => {
     });
 
     const [smartPort, setSmartPort] = useState<SmartPort>();
+    const [unknownStorage, setUnknownStorage] = useState<UnknownStorage>();
```
Comment (iflan) [inline js/components/Drives.tsx:76]: Consider moving these up next to the `useState<Disk2>` line and having all of the `data1`, `data2`, lines follow them all.

File: js/components/Drives.tsx [hunk 3]
```diff
@@ -71,27 +73,31 @@ export const Drives = ({ cpu, io, sectors, enhanced, ready }: DrivesProps) => {
...
 
     const hash = useHash();
 
     useEffect(() => {
-        if (smartPort && disk2) {
+        if (disk2 && unknownStorage) {
```
Comment (iflan) [inline js/components/Drives.tsx:81]: Can `unknownStorage` be set without `disk2` being set?

**Review by iflan — APPROVED** [review]
LGTM. I mostly have questions below because I'm not familiar with the DND APIs.

**Pull request merged**